### ISSUE TITLE
costacoffee_gb spider missing some branches

### ISSUE THREAD

**Issue by rjw62** [alltheplaces/alltheplaces#4235]
Comparing the AllThePlaces data to that in OSM at https://osm.mathmos.net/chains/Q608845/ I was surprised at the number (>200) of OSM objects that were not matched to a branch in the list. You'd expect some former branches to still be listed in OSM, but probably not that many.

I've checked a couple with recent (2022 Q1) FHRS inspections, and I think the spider is not picking up some branches for some reason. For example, the following two branches:

- https://ratings.food.gov.uk/business/en-GB/621062 (1 Isambard Parade, EX4 4BX)
- https://ratings.food.gov.uk/business/en-GB/266946 (1 The Green, TN16 1AS)

can both be found by searching on Costa's website at https://www.costa.co.uk/locations/store-locator/ and are also both present in the JSON feed at https://www.costa.co.uk/api/locations/stores which is used by the spider. However, neither of them appear in the latest AllThePlaces file: https://data.alltheplaces.xyz/runs/2022-11-12-13-31-47/output/costacoffee_gb.geojson

I've had a quick look at the spider code at https://github.com/alltheplaces/alltheplaces/blob/master/locations/spiders/costacoffee_gb.py and the entries at https://www.costa.co.uk/api/locations/stores and didn't spot any obvious reason why they wouldn't be picked up. Anyone have any ideas?

**Comment by Cj-Malone** [comment]
I just ran it locally and both examples showed up. Also the first is precent in [last weeks data](https://data.alltheplaces.xyz/runs/2022-11-05-13-31-55/output/costacoffee_gb.geojson), but the latter isn't... This weeks data should be finished at some point today, so lets see what happens.

Also the storeStatus != TRADING check seems to have no effect on output right now. I'd imagine they are only returning trading stores.

**Comment by Cj-Malone** [comment]
Possibly related https://www.devonlive.com/whats-on/food-drink/exeter-costa-coffee-branch-shuts-7650338

**Comment by Cj-Malone** [comment]
[2022-11-19-13-32-36](https://data.alltheplaces.xyz/runs/2022-11-19-13-32-36/output/costacoffee_gb.geojson) includes both those postcodes, with +69 entries from last week. Could you rerun the conflation with the new data? To see what happens.

Unfortunately, I think the answer is gonna be, some spiders fluctuate, some wildly, and there needs to some kind of normalisation over the last x weeks of data before it gets used in "production".

**Comment by Cj-Malone** [comment]
@rjw62 did you re run your conflation? The latest run had 14,008 items.

**Comment by davidhicks** [comment]
This is possibly addressed by the rewrite of the `costacoffee_gb` spider per #5717. I checked locations returned by the new spider of #5717 against the previous spider and noticed 500 or so fewer locations returned. I checked those locations against the official store finder web page and confirmed those extra 500 locations weren't displaying. I suspect that the previous spider was including closed locations, and the new spider of #5717 is correctly ignoring closed locations.
